src/content/tools/widget-previewer.md

```diff
@@ -0,0 +1,150 @@
+The Flutter Widget Preview requires Flutter version 3.35 or
+higher.
+
+Please be aware that this is an **experimental release**
+available in the Flutter stable channel. The APIs are not
+stable and *will change*. This guide is for the current
+early access version, and you should expect future updates
+to introduce breaking changes.
+:::
+
+## Start the previewer
+
+To start the Flutter Widget Previewer, navigate to your
+Flutter project's root directory and run the following
+command in your terminal. This will launch a local server
+and open a Widget Preview environment in Chrome that
+automatically updates based on changes to your project.
+
+```shell
+flutter widget-preview start
+```
+
+## Preview a widget
+
+After you've started the previewer, to look at a widget,
+you must use the [`@Preview`][] annotation defined in
+`package:flutter/widget_previews.dart`. This annotation
+can be applied to:
+
+*   **Top-level functions** that return a `Widget` or
+    `WidgetBuilder`.  
+*   **Static methods** within a class that return a `Widget` or
+    `WidgetBuilder`.  
+*   **Public Widget constructors and factories** with no
+    required arguments.
+
+Here is a basic example of how to use the
+`@Preview` annotation to preview a `Text` widget:
+
+```dart
+import 'package:flutter/widget_previews.dart';
+import 'package:flutter/material.dart'; // For Material widgets
+
+@Preview(name: 'My Sample Text')
+Widget mySampleText() {
+  return const Text('Hello, World!');
+}
+```
+
+![Sample widget in Flutter Widget Previewer](/assets/images/docs/tools/widget-previewer/widget-previewer.png "Example widget"){:width="50%"}
+
+Each preview instance provides various controls for
+interacting with the previewed widget. From left to right:
+
+*   **Zoom in:** Magnifies the widget in the preview.
+
+*   **Zoom out:** Reduces the magnification of the widget in
+    the preview.
+
+*   **Reset zoom:** Returns the widget preview to its
+    default zoom level.
+
+*   **Toggle between light and dark mode:** Switches the
+    preview's theme between a light and dark color scheme.
+
+*   **Perform a hot restart for the individual preview:**
+    Restarts only the specific widget preview,
+    allowing changes to be applied quickly without
+    restarting the entire application.
+
+For the case where global state has been modified
+(for example, a static initializer has been changed), the
+entire widget previewer can be told to hot restart using the
+button at the top right of the environment.
+
+## Customize a preview
+
+The [`@Preview`][] annotation has several parameters you can
+use to customize the preview:
+
+*   **`name`**: A descriptive name for the preview.
+
+*   **`size`**: Artificial size constraints using a Size
+    object.
+  
+*   **`textScaleFactor`**: A custom font scale.
+
+*   **`wrapper`**: A function that wraps your previewed
+    widget in a specific widget tree (for example, to inject
+    application state into the widget tree via a Provider).
```

Comment:
"A Provider" is a terminology specific to a package that developers might not be familiar with, so consider replacing with a mention of the generic Flutter version:

```suggestion
    application state into the widget tree with a `InheritedWidget `).
```